changelog.d/issue-9777

```diff
@@ -0,0 +1,43 @@
+synopsis: Installing a library with a Custom setup no longer requires building the executable
+packages: Cabal
+prs: #10311
```

Comment:
Okay. what does it mean to skip the changelog entry? do you want me to add this to the changelog right away?

Reply:
I thought the changelog was generated programatically from the entries

Reply:
> what does it mean to skip the changelog entry? do you want me to add this to the changelog right away?

yes please

> I thought the changelog was generated programatically from the entries

It is, but we mostly do that to avoid merge conflicts, which shouldn't be a concern in this case. Also now there are only >=3.14 entries in changelog.d/, and we don't want to mix them up.

Reply:
I agree with Francesco that at this point it's too late to go via changelog-files, and one of the [release-notes documents](https://github.com/haskell/cabal/tree/master/release-notes) should be adjusted retroactively. Not sure 3.12.0.0 or 3.12.0.1: need to double-check with the git history.

Reply:
How will the retroactive change make it into the hackage changelog?
I've updated it now.

Reply:
The Hackage changelogs just point to the changelog files on `master`.

Reply:
Got it. This is ready then, please take a look again.

Reply:
> The Hackage changelogs just point to the changelog files on `master`.

With the changelog files being markdown, they're easily publishable elsewhere too as I'm doing here with an update for Cabal's own website, (as an update to https://www.haskell.org/cabal/).

https://discourse.haskell.org/t/hakyll-is-awesome/9838
https://cabal.blockscope.com/releases/

Reply:
The ability to hotfix the release notes is soo convenient. But we are not overusing it and the notes/changelogs pertain to one release only (and we never modify any but the last one, I think), so snapshots of our release notes shall never get too outdated. Maybe we should state somewhere clearly that the only authoritative source of the changelog is the master branch in the repo. But linking to a files in a moving branch quite clearly indicates that, doesn't it?

Reply:
I think what might be more helpful is documentation that we do this _and_ why. In particular, I personally found the indirection in the Hackage changelogs annoying at first, especially given that if you read on you find older actual changelogs. This would be a fixed preamble in the changelog files.